plugins/MauticCrmBundle/Command/ProcessPipedriveDeletionsCommand.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace MauticPlugin\MauticCrmBundle\Command;
+
+use Doctrine\ORM\EntityManager;
+use Mautic\CoreBundle\Factory\ModelFactory;
+use Mautic\LeadBundle\Entity\Company;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\LeadBundle\Model\CompanyModel;
+use Mautic\LeadBundle\Model\LeadModel;
+use Mautic\PluginBundle\Entity\IntegrationEntity;
+use Mautic\PluginBundle\Helper\IntegrationHelper;
+use MauticPlugin\MauticCrmBundle\Entity\PipedriveDeletion;
+use MauticPlugin\MauticCrmBundle\Integration\PipedriveIntegration;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class ProcessPipedriveDeletionsCommand extends ContainerAwareCommand
```

Comment:
The ContainerAwareCommand and will be removed in Symofny 5. As this is a feature request it will go directly to the 5.x branch so please do not use any deprecated code. Use DI. See https://github.com/mautic/mautic/pull/11041

Looking at this Command class it looks like it does too much. Commands and controllers should be just thin layers calling services that do the heavy lifting. When you look at some old commands and controllers you can see that we've learned that the hard way :) Please try to divide it into one or more services so the code could be reused and to achieve SOLID principles. Especially the [Single Responsibility](https://en.wikipedia.org/wiki/Single-responsibility_principle).